tests/gateways/test_repodata_gateway.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+Strongly related to subdir_data / test_subdir_data.
+"""
+
+from __future__ import annotations
+
+import sys
+
+import pytest
+
+from conda.base.context import conda_tests_ctxt_mgmt_def_pol
+from conda.common.io import env_vars
+from conda.exceptions import (
+    CondaDependencyError,
+    CondaHTTPError,
+    CondaSSLError,
+    ProxyError,
+    UnavailableInvalidChannel,
+)
+from conda.gateways.connection import HTTPError, InvalidSchema, RequestsProxyError, SSLError
+from conda.gateways.repodata import RepodataIsEmpty, conda_http_errors
+
+
+def test_coverage_conda_http_errors():
```

Comment:
This might be useful to decompose into individual test functions

Reply:
I think the problematic `test_subdir_data_coverage` makes more sense to decompose since it is doing a couple of unrelated things, but not this one which does the same thing ten times.  This test could be decomposed into a loop over an (exception to raise, expected) values.

Reply:
Yeah, that's fair.